File: src/contexts/chat/infrastructure/mock/in-memory-models.gateway.ts
import { IModelsGateway } from '@contexts/chat/application/ports/models.gateway';
import { ModelAggregate } from '@contexts/chat/domain/aggregates/model.aggregate';
import { ModelNotFoundException } from '@contexts/chat/domain/exceptions/model-not-found.exception';
import { Injectable } from '@nestjs/common';

/**
 * Stands in for nexora-models: a hardcoded catalog of a single model,
 * until nexora-models exists.
 */
@Injectable()
export class InMemoryModelsGateway implements IModelsGateway {
  private readonly model = new ModelAggregate('nexora-mock-llama-3.1-8b');

  async resolve(name: string): Promise<ModelAggregate> {
    if (this.model.name !== name) {
      throw new ModelNotFoundException(name, this.model.name);
    }
    return this.model;
  }
}

File: src/main/java/com/sisqueslabs/nexora/api/contexts/chat/infrastructure/mock/InMemoryModelsGateway.java
package com.sisqueslabs.nexora.api.contexts.chat.infrastructure.mock;

import org.springframework.stereotype.Component;

import com.sisqueslabs.nexora.api.contexts.chat.application.port.ModelsGateway;
import com.sisqueslabs.nexora.api.contexts.chat.domain.entities.Model;
import com.sisqueslabs.nexora.api.contexts.chat.domain.exceptions.ModelNotFoundException;

/**
 * Stands in for nexora-models: a hardcoded catalog of a single model,
 * until nexora-models exists.
 */
@Component
public class InMemoryModelsGateway implements ModelsGateway {

    private final Model model = new Model("nexora-mock-llama-3.1-8b");

    @Override
    public Model resolve(String name) {
        if (!model.name().equals(name)) {
            throw new ModelNotFoundException(name, model.name());
        }
        return model;
    }
}
